File: 35-search-insert-position/search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let low = 0;
    let high = nums.length - 1;

    while(low <= high) {
        let mid = Math.floor((low + high) / 2);
        let midVal = nums[mid];
        if(target === midVal) {
            return mid;
        } else if(target > midVal) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    return low;
};

File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int low = 0;
        int high = nums.length - 1;
        while (low <= high) {
            int mid = (low + high) / 2;
            int midValue = nums[mid];
            if (target == midValue) {
                return mid;
            } else if (target > midValue) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return low;
    }
}